openfl-tutorials/experimental/Keras_MNIST_with_CPU.ipynb

```diff
@@ -0,0 +1,320 @@
+    "# To turn output vectors into probability vector, a function called Softmax is often used \n",
+    "# as the last activation function in a classification neural network. \n",
+    "# For example, softmax([−1,1,2])=[0.035,0.25,0.705].\n",
+    "model = Sequential([\n",
+    "    Conv2D(filters=32, kernel_size=(3, 3), activation=\"relu\", input_shape=(28, 28, 1)),\n",
+    "    MaxPool2D(), \n",
+    "    Flatten(), # Converts the multi-dimensional feature map into a one-dimensional vector.\n",
+    "    Dense(512, activation=\"relu\"), # A fully connected layer with 512 neurons, used to learn higher-level abstract features and representations.\n",
+    "    Dropout(0.2), # Randomly sets 20% of the neurons' outputs to zero during training, avoids relying on a specific neuron.\n",
+    "    Dense(512, activation=\"relu\"), # A fully connected layer with 512 neurons, used to learn higher-level abstract features and representations.\n",
+    "    Dropout(0.2), # Randomly sets 20% of the neurons' outputs to zero during training, avoids overfitting.\n",
+    "    Dense(nb_classes, activation=\"softmax\"), # To turn output vectors into probability vector,\n",
+    "])\n",
+    "\n",
+    "model.compile(optimizer=\"adam\", loss=\"categorical_crossentropy\", metrics=[\"accuracy\"])\n",
+    "print(model.summary())\n",
+    "\n",
+    "\n",
+    "def FedAvg(models):\n",
+    "    new_model = models[0]\n",
+    "    state_dicts = [model.weights for model in models]\n",
+    "    state_dict = new_model.weights\n",
+    "    for idx, _ in enumerate(models[1].weights):\n",
+    "        state_dict[idx] = np.sum(np.array([state[idx]\n",
+    "                                 for state in state_dicts], dtype=object), axis=0) / len(models)\n",
+    "    new_model.set_weights(state_dict)\n",
+    "    return new_model\n",
+    "\n",
+    "def inference(model, test_loader, batch_size):\n",
+    "    x_test, y_test = test_loader\n",
+    "    loss, accuracy = model.evaluate(\n",
+    "        x_test,\n",
+    "        y_test,\n",
+    "        batch_size=batch_size,\n",
+    "        verbose=0\n",
+    "    )\n",
+    "    accuracy_percentage = accuracy * 100\n",
+    "    print(f\"Test set: Avg. loss: {loss}, Accuracy: {accuracy_percentage:.2f}%\")\n",
+    "    return accuracy"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "72bfdbed",
+   "metadata": {},
+   "source": [
+    "Initialize the aggregator and collaborators. Edit collaborator_names to add/remove collaborators."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3a5964c0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from openfl.experimental.interface import FLSpec, Aggregator, Collaborator\n",
+    "from openfl.experimental.runtime import LocalRuntime\n",
+    "from openfl.experimental.placement import aggregator, collaborator\n",
+    "\n",
+    "agg = Aggregator()\n",
+    "\n",
+    "collaborator_names = [\"C1\", \"C2\"]\n",
```

Comment:
As I suggested to Kush also, let's call these after big cities, to highlight the geo-distributed aspect of FL!

Reply:
Renamed the collaborators.